src/System.Diagnostics.Tracing.Telemetry/src/System/Diagnostics/Tracing/TelemetryListener.cs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// A TelemetryListener is something that forwards on events written with TelemetrySource.
+    /// It is an IObservable (has Subscribe method), and it also has a Subscribe overload that
+    /// lets you specify a 'IsEnabled' predicate that users of TelemetrySource will use for 
+    /// 'quick checks'.   
+    /// 
+    /// The item in the stream is a KeyValuePair[string, object] where the string is the name
+    /// of the telemetry item and the object is the payload (typically an anonymous type).  
+    /// 
+    /// There may be many TelemetryListeners in the system, but we encourage the use of
+    /// The TelemetrySource.DefaultSource which goes to the TelemetryListener.DefaultListener.
+    /// 
+    /// If you need to see 'everything' you can subscribe to the 'AllListeners' event that
+    /// will fire for every live TelemetryListener in the appdomain (past or present). 
+    /// </summary>
+    public class TelemetryListener : TelemetrySource, IObservable<KeyValuePair<string, object>>, IDisposable
+    {
+        /// <summary>
+        /// This is the TelemetryListener that is used by default by the class library.   
+        /// Generally you don't want to make your own but rather have everyone use this one, which
+        /// ensures that everyone who wished to subscribe gets the callbacks.  
+        /// The main reason not to us this one is that you WANT isolation from other 
+        /// events in the system (e.g. multi-tenancy).  
+        /// </summary>
+        public static TelemetryListener DefaultListener { get { return s_default; } }
+
+        /// <summary>
+        /// When you subscribe to this you get callbacks for all NotificationListeners in the appdomain
+        /// as well as those that occurred in the past, and all future Listeners created in the future. 
+        /// </summary>
+        public static event Action<TelemetryListener> AllListeners
```

Comment:
I would rather have it as a property that returns List<TelemetryListener> + an event called ListenersChanged. The reason is that our conventions don't allow events typed as Action<T>. They need to be typed as a subtype of EventHandler.
Having said that, I think it would be fine to do this API fix post the initial commit.